powersimdata/data_access/data_access.py

```diff
@@ -56,18 +56,26 @@ def remove(self, target, recursive=False, force=False):
         command = CommandBuilder.remove(target, recursive, force)
         return self.execute_command(command)
 
+    def _exists(self, filepath):
+        """Return whether the file exists
+
+        :param str filepath: the path to the file
+        :return: (*bool*) -- whether the file exists
+        """
+        raise NotImplementedError
```

Comment:
I presume the reason we put this down here is for situations in the future, we have other classes that inherit `DataAccess` without overwriting `_exists` method?

Reply:
Yep, that's pretty much it. I'd say a couple related reasons are 1) make it so that `self._exists` is defined in the context we're referencing it (make editor diagnostics happy), and 2) it's a form of documentation, so you know what needs to be implemented in a subclass.